src/components/Redirect.jsx

```diff
@@ -0,0 +1,80 @@
+// @flow
+import React from 'react';
+import withSideEffect from 'react-side-effect';
+import RouterRedirect from 'react-router-dom/Redirect';
+
+const testForExternal = (to: RouterTo): boolean => {
+	if (to instanceof URL) {
+		return true;
+	}
+	if (typeof to === 'string') {
+		return to.startsWith('http');
+	}
+	return false; // this is a React Router 'location'
+};
+
+type RouterTo = string | LocationShape | URL;
+type RedirectProps = {
+	to: RouterTo,
+	push?: boolean,
+};
+
+/*
+ * A routing component that, when rendered, will redirect the application to
+ * another internal route _or_ external URL. On the server, the redirect will
+ * be handled with a 302 redirect HTTP response to the browser. On the client,
```

Comment:
What if we need to do a 301? I'm soon going to be migrating a page that will live at 1 location to a new one, ideally i would use the Redirect to send people on their way but would be good to have it recorded as 301 so SEO picks that up